rosbag2_py/package.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>rosbag2_py</name>
+  <version>0.2.4</version>
+  <description>Python API for rosbag2</description>
+  <maintainer email="jacob@openrobotics.org">Jacob Perron</maintainer>
+  <maintainer email="ros-tooling@googlegroups.com">ROS Tooling Working Group</maintainer>
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>ament_cmake_ros</buildtool_depend>
+
+  <depend>pybind11_vendor</depend>
```

Comment:
We should confirm, but I think we can rely on the [existing rosdep key](https://github.com/ros/rosdistro/blob/ce7f25888bae5a321ed8ebd53848ad6e85503b5e/rosdep/base.yaml#L5274) for Linux.

```suggestion
  <depend>pybind11-dev</depend>
```

For macOS and Windows, we should update our CI script and install instructions to install via pip: https://pypi.org/project/pybind11/

e.g. `python -m pip install pybind11`

Can you try updating the dependency key here? I can update https://github.com/ros2/ci to ensure that pybind11 is installed for macOS and Windows.

Reply:
That worked for me on Linux!

Reply:
I've pushed changes to ros2/ci for installing pybind11 with pip on macOS and Windows: https://github.com/ros2/ci/commit/7782acc904eb72b2874cafa9b5b8851b0d64e21a

I've triggered CI to see how it goes.

Reply:
Turns out the pip package does not ship with the necessary cmake config. Relevant upstream issue: https://github.com/pybind/pybind11/issues/1628

I think the best course of action is to create a `pybind11_cmake_module` package providing one or add it to `rosbag2_py`.

Reply:
I looked into this for a while and found out a few things. First, I think my test above was not thorough. I did not place a `COLCON_IGNORE` in my pybind11_vendor package, so even though I did not depend on it, colcon was likely still using it to find pybind. That was why it worked. Once I put a `COLCON_IGNORE` in pybind11_vendor, colcon fails to find pybind11Config.cmake.

Pybind only supplies the .cmake file when it is installed with Conda https://github.com/pybind/pybind11/issues/1379 There's been a lot of complaints but they aren't interested in fixing it. I got `pybind11Config.cmake`, `pybind11Tools.cmake`, `pybind11Targets.cmake`, `FindPythonLibsNew.cmake` from a Conda install of pybind, and the `pybind11Config.cmake.in` file which generates the `pybind11Config.cmake` from https://github.com/pybind/pybind11/blob/master/tools/pybind11Config.cmake.in , and I was able to hook all these files up. But the remaining problem is that I can't get `@PACKAGE_INIT@` to expand to the pybind11 installation directory. I don't see how it can ever know where that path is on the user's local machine. It expands to `../..`, which is correct if run within pybind11, but at the point it's run within rosbag2_py, find(pybind11) doesn't work yet, so we don't have the installation path of pybind11 to expand that to. That path is further used to set the `pybind11_INCLUDE_DIRS`, which is what my colcon is complaining right now - can't find `pybind/pybind.h`.

So, I've gotten a lot farther, but I don't think there's a way forward unless there's a way to figure out where pybind11 is installed locally - before pybind11Config.cmake is generated. All this is to say, I propose we go back to using `pybind11_vendor`.

Reply:
Switching back to the vendor package sounds okay to me. Thanks for looking into it :bow: 

Please add the vendor package to the [ros2.repos](https://github.com/ros2/ros2/blob/master/ros2.repos) file and trigger CI with it and this PR.